File: oapp/services/restaurant/internal/models/store.go
package models

import "time"

// Store represents a restaurant store
type Store struct {
	StoreID     string    `json:"store_id" db:"store_id"`
	Name        string    `json:"name" db:"name"`
	Description *string   `json:"description,omitempty" db:"description"`
	LogoURL     *string   `json:"logo_url,omitempty" db:"logo_url"`
	Phone       string    `json:"phone" db:"phone"`
	CreatedAt   time.Time `json:"created_at" db:"created_at"`
	UpdatedAt   time.Time `json:"updated_at" db:"updated_at"`
}

// CreateStoreRequest represents the request body for creating a store
type CreateStoreRequest struct {
	Name        string  `json:"name" validate:"required,max=255"`
	Description *string `json:"description,omitempty" validate:"omitempty,max=1000"`
	LogoURL     *string `json:"logo_url,omitempty"`
	Phone       string  `json:"phone" validate:"required,max=20"`
}

// UpdateStoreRequest represents the request body for updating a store
type UpdateStoreRequest struct {
	Name        *string `json:"name,omitempty" validate:"omitempty,max=255"`
	Description *string `json:"description,omitempty" validate:"omitempty,max=1000"`
	LogoURL     *string `json:"logo_url,omitempty"`
	Phone       *string `json:"phone,omitempty" validate:"omitempty,max=20"`
}

File: oapp/DeliveryXOApp/DeliveryXOApp/Models/Store.swift
//
//  Store.swift
//  DeliveryXOApp
//
//  Created on 2025-10-21.
//

import Foundation

/// Represents a store/restaurant in the system
struct Store: Codable, Identifiable {
    let id: String
    let name: String
    let email: String?
    let phone: String?
    let description: String?
    let logoUrl: String?
    let createdAt: Date
    let updatedAt: Date

    enum CodingKeys: String, CodingKey {
        case id = "store_id"
        case name
        case email
        case phone
        case description
        case logoUrl = "logo_url"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

/// Request model for creating a new store
struct CreateStoreRequest: Codable {
    let name: String
    let email: String
    let phone: String?
    let description: String?
    let logoUrl: String?

    enum CodingKeys: String, CodingKey {
        case name
        case email
        case phone
        case description
        case logoUrl = "logo_url"
    }
}
